Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt

```diff
@@ -0,0 +1,54 @@
+package de.rki.coronawarnapp.ui.coronatest.rat.profile.list.items
+
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import coil.loadAny
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.ProfileCardItemBinding
+import de.rki.coronawarnapp.ui.coronatest.rat.profile.list.ProfileListAdapter
+import de.rki.coronawarnapp.ui.coronatest.rat.profile.qrcode.PersonProfile
+import de.rki.coronawarnapp.util.coil.loadingView
+import de.rki.coronawarnapp.util.qrcode.coil.CoilQrCode
+
+class ProfileCard(parent: ViewGroup) : ProfileListAdapter.ItemVH<ProfileCard.Item, ProfileCardItemBinding>(
+    layoutRes = R.layout.home_card_container_layout,
+    parent
+) {
+
+    override val viewBinding = lazy {
+        ProfileCardItemBinding.inflate(layoutInflater, itemView.findViewById(R.id.card_container), true)
+    }
+
+    override val onBindData: ProfileCardItemBinding.(
+        item: Item,
+        payloads: List<Any>
+    ) -> Unit = { item, payloads ->
+        val curItem = payloads.filterIsInstance<Item>().lastOrNull() ?: item
+        val fullName = buildString {
+            if (curItem.personProfile.profile != null) {
+                append(curItem.personProfile.profile.firstName)
```

Comment:
You need to check whether the names are bank here, otherwise it will look ugly with the needless space as a separator

Reply:
The profile should not be nullable in the list

Reply:
Alright, I was using this `PersonProfile` data class that I found in `RATProfileQrCodeFragmentViewModel` where both the profile and the qrCode are nullable, but I can actually pass the profile and the qrCode directly to avoid checking for null here.

Reply:
you can just use trim()

Reply:
and make the profile not nullable, cannot be null in that fragment, just use Profile() as a fallback